### DISCUSSION TITLE
Dependency between signals

### DISCUSSION THREAD

**Discussion by saona-raimundo** [leptos-rs/leptos#1271] [Q&A] [answered]
Thanks for all the work!!

Assume you have a big application and you want to debug/study/test the dependency between the different signals used. How would go about it?

In particular, 
1) Is there a way to print some sort of (descendency) tree of the dependent signals from a signal?
2) Is there a way to print some sort of (ascendency) tree of the signals I depend on?

**Comment by gbj** [comment, marked as answer]
So the answer to the questions is: Not really! i.e., there's not existing tooling that makes it easy to examine these relationships.

It would be fairly straightforward to add a minimal version of this with the data that the reactive system already includes.

Leptos does have quite thorough `tracing` integration, and [`tracing-subscriber-wasm`](https://crates.io/crates/tracing-subscriber-wasm) is built for it specifically.
